File: lab-files/SIR.cs
public class SIRModel
{
    private double susceptible;
    private double infected;
    private double recovered;
    private double totalPopulation;
    private double beta;
    private double gamma;

    public SIRModel(double totalPopulation, double initialInfected, double initialRecovered, double beta, double gamma)
    {
        this.totalPopulation = totalPopulation;
        this.infected = initialInfected;
        this.recovered = initialRecovered;
        this.susceptible = totalPopulation - initialInfected - initialRecovered;
        this.beta = beta;
        this.gamma = gamma;
    }

    public void Calculate(double timeStep, int steps)
    {
        for (int i = 0; i < steps; i++)
        {
            double newInfected = timeStep * beta * susceptible * infected / totalPopulation;
            double newRecovered = timeStep * gamma * infected;

            susceptible -= newInfected;
            infected += newInfected - newRecovered;
            recovered += newRecovered;

            Console.WriteLine($"Step {i}: Susceptible = {susceptible}, Infected = {infected}, Recovered = {recovered}");
        }
    }
}

File: lab-files/SIR.py
class SIRModel:
    def __init__(self, total_population, initial_infected, initial_recovered, beta, gamma):
        self.total_population = total_population
        self.infected = initial_infected
        self.recovered = initial_recovered
        self.susceptible = total_population - initial_infected - initial_recovered
        self.beta = beta
        self.gamma = gamma

    def calculate(self, time_step, steps):
        for i in range(steps):
            new_infected = time_step * self.beta * self.susceptible * self.infected / self.total_population
            new_recovered = time_step * self.gamma * self.infected

            self.susceptible -= new_infected
            self.infected += new_infected - new_recovered
            self.recovered += new_recovered

            print(f"Step {i}: Susceptible = {self.susceptible}, Infected = {self.infected}, Recovered = {self.recovered}")